protocol-designer/src/labware-ingred/utils.ts

```diff
@@ -1,16 +1,45 @@
-import { RobotType, getDeckDefFromRobotType } from '@opentrons/shared-data'
+import {
+  FIXED_TRASH_ID,
+  getDeckDefFromRobotType,
+  MOVABLE_TRASH_ADDRESSABLE_AREAS,
+  WASTE_CHUTE_ADDRESSABLE_AREAS,
+} from '@opentrons/shared-data'
+import { COLUMN_4_SLOTS } from '@opentrons/step-generation'
 import { getSlotIsEmpty } from '../step-forms/utils'
-import { InitialDeckSetup } from '../step-forms/types'
-import { DeckSlot } from '../types'
+import { getStagingAreaAddressableAreas } from '../utils'
+import type { RobotType, CutoutId } from '@opentrons/shared-data'
+import type { InitialDeckSetup } from '../step-forms/types'
+import type { DeckSlot } from '../types'
 
 export function getNextAvailableDeckSlot(
   initialDeckSetup: InitialDeckSetup,
   robotType: RobotType
 ): DeckSlot | null | undefined {
   const deckDef = getDeckDefFromRobotType(robotType)
-  return deckDef.locations.addressableAreas.find(slot =>
-    getSlotIsEmpty(initialDeckSetup, slot.id)
-  )?.id
+
+  return deckDef.locations.addressableAreas.find(slot => {
+    const cutoutIds = Object.values(initialDeckSetup.additionalEquipmentOnDeck)
+      .filter(ae => ae.name === 'stagingArea')
+      .map(ae => ae.location as CutoutId)
+    const stagingAreaAddressableAreaNames = getStagingAreaAddressableAreas(
+      cutoutIds
+    )
+    const addressableAreaName = stagingAreaAddressableAreaNames.find(
+      aa => aa === slot?.id
```

Comment:
nit
from the code, I think slot isn't `undefined` also because of L31
```suggestion
      aa => aa === slot.id
```